server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailGet.scala

```diff
@@ -212,8 +217,21 @@ case class EmailGetResponse(accountId: AccountId,
                             list: List[Email],
                             notFound: EmailNotFound)
 
-case class SpecificHeaderRequest(headerName: NonEmptyString, property: String) {
-  def retrieveHeader(message: Message): (String, Option[EmailHeaderValue]) = (headerName,
-    Option(message.getHeader.getField(property))
-    .map(field => EmailHeaderValue.from(field)))
+case class SpecificHeaderRequest(headerName: NonEmptyString, property: String, parseOption: Option[String]) {
+  def retrieveHeader(message: Message): (String, Option[EmailHeaderValue]) = parseOption match {
+    case Some(parseOption) => retrieveHeaderWithParseOption(message, property, parseOption)
+    case None => (headerName,
+      Option(message.getHeader.getField(property))
+        .map(field => EmailHeaderValue.from(field)))
+  }
+
+  private def retrieveHeaderWithParseOption(message: Message, property: String, parseOption: String): (String, Option[EmailHeaderValue]) = {
```

Comment:
Let me suggest...

To have EmailHeaderValue a sealed trait, then we could have RawHeaderValue, TextHeaderValue etc...

Each one would have a "fromField" factory method in an object.

Then we could:

```
val field: Option[Field] = Option(message.getHeader.getField(property))

parseOption match {
   case AsRow => (headerName, field.map(RawHeaderValue.from))
   case AsText => (headerName, field.map(TextHeaderValue.from))
   case AsDate => (headerName, field.map(DateHeaderValue.from))
   ...
}
```

Reply:
Use a case object to parseOption

```
sealed trait ParseOption
case object AsRaw extends ParseOption
...
```